### PR TITLE
fix: update json-schema

### LINKED ISSUE #22249: Allow regex:* strings in versioning for JSON schema
### What would you like Renovate to be able to do?

Allow the `regex:` string pattern in JSON schema.

### If you have any ideas on how this should be implemented, please tell us here.

See https://github.com/renovatebot/renovate/discussions/22247#discussioncomment-5911999

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- Allow `^regex:.*` as `versioning`
- Add description to global array objects (like `packageRules` or `regexManagers`)

## Context
- fixes #22249

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
```diff
1054a1055,1058
>               "description": {
>                 "type": "string",
>                 "description": "A custom description"
>               },
1553a1558,1561
>               "description": {
>                 "type": "string",
>                 "description": "A custom description"
>               },
2069a2078,2081
>               "description": {
>                 "type": "string",
>                 "description": "A custom description"
>               },
2329a2342,2345
>               "description": {
>                 "type": "string",
>                 "description": "A custom description"
>               },
2748,2784c2764,2808
<       "enum": [
<         "aws-machine-image",
<         "azure-rest-api",
<         "bazel-module",
<         "cargo",
<         "composer",
<         "conan",
<         "deb",
<         "debian",
<         "docker",
<         "git",
<         "go-mod-directive",
<         "gradle",
<         "hashicorp",
<         "helm",
<         "hermit",
<         "hex",
<         "ivy",
<         "kubernetes-api",
<         "loose",
<         "maven",
<         "nixpkgs",
<         "node",
<         "npm",
<         "nuget",
<         "pep440",
<         "perl",
<         "poetry",
<         "python",
<         "redhat",
<         "regex",
<         "rez",
<         "ruby",
<         "semver",
<         "semver-coerced",
<         "swift",
<         "ubuntu"
---
>       "oneOf": [
>         {
>           "enum": [
>             "aws-machine-image",
>             "azure-rest-api",
>             "bazel-module",
>             "cargo",
>             "composer",
>             "conan",
>             "deb",
>             "debian",
>             "docker",
>             "git",
>             "go-mod-directive",
>             "gradle",
>             "hashicorp",
>             "helm",
>             "hermit",
>             "hex",
>             "ivy",
>             "kubernetes-api",
>             "loose",
>             "maven",
>             "nixpkgs",
>             "node",
>             "npm",
>             "nuget",
>             "pep440",
>             "perl",
>             "poetry",
>             "python",
>             "redhat",
>             "regex",
>             "rez",
>             "ruby",
>             "semver",
>             "semver-coerced",
>             "swift",
>             "ubuntu"
>           ]
>         },
>         {
>           "type": "string",
>           "pattern": "^regex:"
>         }
```

**Comment by viceice** [comment]
it adds the description field to the schema, which we already heavily use to describe package rules. so the editor can suggest that field now.

**Comment by viceice** [comment]
i can change the description of the description property, it seems to confusing 😬

**Comment by viceice** [comment]
> @viceice I don't understand this "a custom description" change. Here's an example:
> 
> ```json
>     "hostRules": {
>       "description": "Host rules/configuration including credentials.",
>       "type": "array",
>       "items": {
>         "allOf": [
>           {
>             "type": "object",
>             "properties": {
>               "description": {
>                 "type": "string",
>                 "description": "A custom description"
>               },
>               "abortIgnoreStatusCodes": {
>                 "description": "A list of HTTP status codes safe to ignore even when `abortOnError=true`.",
>                 "type": "array",
>                 "items": {
>                   "type": "number"
>                 }
>               },
> ```

`description` is added for child objects of these root properties:
image

File: tools/docs/schema.ts [hunk 1]
```diff
@@ -83,7 +88,12 @@ function addChildrenArrayInParents(): void {
         allOf: [
           {
             type: 'object',
-            properties: {},
+            properties: {
+              description: {
+                type: 'string',
+                description: 'A custom description',
```
Comment (viceice) [inline tools/docs/schema.ts:94]:
```suggestion
```

Should we remove this, or change the text.

Comment (viceice) [reply tools/docs/schema.ts:94]: image

Comment (rarkins) [reply tools/docs/schema.ts:94]: I think I understand it now, maybe it's ok. It's more understandable in the IDE than in the JSON schema itself. Maybe "A custom description for this configuration object"

Comment (viceice) [reply tools/docs/schema.ts]: done

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**